src/_pytest/capture.py

```diff
@@ -496,6 +497,7 @@ def suspend_capturing(self, in_=False):
             self._in_suspended = True
 
     def resume_capturing(self):
+        assert self._state in ("suspended", "started"), self._state
```

Comment:
i wonder in what situation resuming from started is valid,
naive thinking makes me wonder if we might want to ensure its always suspended

Reply:
@RonnyPfannschmidt thanks for looking into it, but it's also just WIP, and unlikely to stay - I want to see what tests are hitting that eventually.
You might want to review / approve one of my other non-WIP PRs instead.. ;)